### PR TITLE
CoreData Implementation - Yannic Borgfeld

### PR DESCRIPTION
**Pull request by Cronay**
For the sake of trying it out I implemented the failable specs in a separate [branch](https://github.com/Cronay/ios-lead-essentials-feed-store-challenge/tree/failable_specs) but then decided it's not worth merging into the final solution because the test details leaking into production code were to big I think.

### CODE REVIEW

File: FeedStoreChallenge/CoreDataFeedStore/CoreDataFeedStore.swift [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import Foundation
+import CoreData
+
+public final class CoreDataFeedStore: FeedStore {
+
+    private let persistentContainer: NSPersistentContainer
+    private let context: NSManagedObjectContext
+
+    public init(storeURL: URL) throws {
+        let managedObjectModel = try NSManagedObjectModel.loadFeedStoreModel()
```
Comment (caiozullo) [inline FeedStoreChallenge/CoreDataFeedStore/CoreDataFeedStore.swift:18]: Every time the `CoreDataFeedStore` is instantiated, it creates a new instance of the `NSManagedObjectModel`. But loading a `NSManagedObjectModel` more than once generates console warnings such as:

```
[error] warning: Multiple NSEntityDescriptions claim the NSManagedObject subclass 'ManagedCache' so +entity is unable to disambiguate.
```

You can solve it by making a static `NSManagedObjectModel` instance, so it's loaded *only once* lazily.

**Review by caiozullo — COMMENTED** [review]
Hi @Cronay,

Congratulations, you've passed the challenge! 🥇

- All tests are passing, so your implementation complies with the specs.
- You broke down the implementation in small commits.
- Bonus points for the Integration Tests, Memory Leak Tracking, and Failable Specs.

I left only one comment regarding a console warning. Feel free to push a fix.

Great job ✅

File: FeedStoreChallenge/CoreDataFeedStore/CoreDataFeedStore.swift (hunk 1 above)
Comment (Cronay) [reply FeedStoreChallenge/CoreDataFeedStore/CoreDataFeedStore.swift]: Thanks for the heads up, I fixed it 👍

**Comment by caiozullo** [comment]
Nice one!

**Pull request closed without merging**